sqllineage/core/parser/sqlfluff/extractors/create_insert.py

```diff
@@ -110,6 +111,34 @@ def extract(
                     else:
                         holder.add_write(Path(escape_identifier_name(segment.raw)))
                 tgt_flag = False
+
+                if statement.type == "insert_statement" and holder.write:
+                    sub_segments = list_child_segments(
+                        segment=statement_child[seg_idx + 1]
+                    )
+                    if not all(
+                        sub_segment.type in ["column_reference", "column_definition"]
+                        for sub_segment in sub_segments
+                    ):
+                        tgt_tab = list(holder.write)[0]
+                        if isinstance(tgt_tab, Table) and tgt_tab.schema:
+                            col_list = [
+                                Column(
+                                    name=col.raw_name,
+                                    source_columns=[
+                                        ColumnQualifierTuple(
+                                            column=col.raw_name, qualifier=None
+                                        )
+                                    ],
+                                )
+                                for col in self.metadata_provider.get_table_columns(
+                                    table=Table(
+                                        name=tgt_tab.raw_name,
+                                        schema=tgt_tab.schema.raw_name,
+                                    )
+                                )
```

Comment:
```python
                                for col in self.metadata_provider.get_table_columns(
                                    tgt_tab,
                                )
```

This is more concise.

Reply:
yes， you are right. 
i will modify and redo submit